main.go

```diff
@@ -53,6 +53,10 @@ func _main() int {
 	discoverController := k8sapi.NewController(kubeClient)
 	go discoverController.DiscoverK8SPods()
```

Comment:
If you want to wait for pods to be discovered before moving on, then why not just move the `DiscoverK8SPods()` call into the `NewController` function and perform this initialization in the controller's construction instead of in a separate goroutine here? Wouldn't that be simpler?

Reply:
I thought briefly about this before I made this change. This is used for DiscoveryController today but might be a too strong assumption to say this would be the case going forward. I see this logic to be equivalent to an initialization logic, why start the RPC server or any other process when the required initialization is not done.